### DISCUSSION TITLE
Variants with custom logic?

### DISCUSSION THREAD

**Discussion by JoeVanGundy** [chakra-ui/chakra-ui#8382] [Q&A] [answered]
We're trying to formulate design conventions around how we handle features/logic Chakra doesn't have.
For example, we want to have a variant for `Tabs` called "card" that looks like what's in the screenshot.

I would love to just do something like:
`<Tabs variant="card">`

I can't find any docs on best practices for this.
Would i just need to completely recreate Chakra's Tabs components as a custom component that supports all of the same props/variants/sizes, or is there some easier way?

Alternatively, is the suggestion instead to not extend Chakra's components, and instead create a single custom component for this one purpose? `<BigRadioTabs>`?

Totally stuck on this so any advice, docs, threads would be super appreciated!
Thank you!   

![image](https://github.com/chakra-ui/chakra-ui/assets/8807246/d85e2ce5-e223-44b2-ae77-dd2bd3a5d782)

**Comment by vitalijalbu** [comment]
which website is that??

**Comment by segunadebayo** [comment, marked as answer]
Add a `card` variant on the Tabs slot recipe. You do not fork Tabs.

```ts
defineSlotRecipe({
  slots: tabsAnatomy.keys(),
  variants: {
    variant: {
      card: { root: { /* ... */ }, trigger: { /* ... */ } },
    },
  },
})
```

Then `<Tabs.Root variant="card">`.
